app/Jobs/RockTheVote/ImportRockTheVoteRecord.php

```diff
@@ -203,6 +219,23 @@ private function updateUserIfChanged($user)
         info('Updated user', ['user' => $user->id, 'voter_registration_status' => $this->userData['voter_registration_status']]);
     }
 
+    /**
+     * @param array $post
+     * @return void
+     */
+    private function updatePostIfChanged($post)
+    {
+        if (! self::shouldUpdateStatus($post['status'], $this->postData['status'])) {
+            info('No changes to update for post', ['post' => $post['id']]);
+
+            return;
+        }
+
+        app(Rogue::class)->updatePost($post['id'], ['status' => $this->postData['status']]);
+
+        info('Updated post', ['post' => $post['id'], 'status' => $this->postData['status']]);
+    }
```

Comment:
I think we could keep this method named `updatePost()` and since currently used in only one place move the logic to check if it should update back where line 81 is (`$this->updatePostIfChanged($post);`). Then this method can solely deal w/ updating a post.

Alternatively, having to simpler methods, `updatePostIfChanged()` which does the check and if needs to update then calls `updatePost()`, so each method has a specific responsibility. Allows the `updatePost()` to be called without needing to do the check if that's every a thing.

Reply:
I prefer keeping the check if it should update in here, so we don't have a nested if/else from where it was removed. Also keeps things consistent with the `updateUserIfChanged`, which will soon get more complex once we need to look for changes to SMS status and potentially add an existing user's mobile number.

As far as having two different methods `updatePostIfChanged` and `updatePost` -- having `updatePost` be called without needing to do the check won't ever be a thing because we only update it from this one job.  We could split out the methods at that point if we'd ever need to.

Reply:
Gotcha, ok.

> which will soon get more complex

However this is what I want to help mitigate. The more complex these functions get the harder they are to test (or require more complex tests), which is why I'm always more in favor of smaller single use functions as much as possible.

Reply:
I definitely hear you -- I'm just weary of all if/else logic happening within the `handle. I'll definitely keep this in mind once it's time to work on updating the SMS preferences. Thanks for reviewing so thoroughly!

Reply:
Ya, defs. Lots of if/else logic is not ideal either; there are ways around it but likely require a little refactoring! :)